Content.Server/Health/BodySystem/Mechanism/DroppedMechanismComponent.cs

```diff
@@ -8,32 +8,184 @@
             if (Owner.TryGetComponent<SpriteComponent>(out SpriteComponent component))
             {
                 component.LayerSetRSI(0, data.RSIPath);
                 component.LayerSetState(0, data.RSIState);
             }
         }
+
+        void IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (eventArgs.Target == null)
+                return;
+
+            CloseAllSurgeryUIs();
+            _optionsCache.Clear();
+            _performerCache = null;
+            _bodyManagerComponentCache = null;
+
+            if (eventArgs.Target.TryGetComponent<BodyManagerComponent>(out BodyManagerComponent bodyManager))
+            {
+                SendBodyPartListToUser(eventArgs, bodyManager);
+            }
+            else if (eventArgs.Target.TryGetComponent<DroppedBodyPartComponent>(out DroppedBodyPartComponent droppedBodyPart))
+            {
+                if (droppedBodyPart.ContainedBodyPart == null)
+                {
+                    Logger.Debug("Installing a mechanism was attempted on an IEntity with a DroppedBodyPartComponent that doesn't have a BodyPart in it!");
+                    throw new InvalidOperationException("A DroppedBodyPartComponent exists without a BodyPart in it!");
+                }
+                if (!droppedBodyPart.ContainedBodyPart.TryInstallDroppedMechanism(this))
+                {
+                    _sharedNotifyManager.PopupMessage(eventArgs.Target, eventArgs.User, "You can't fit it in!");
+                }
+            }
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            //This is a temporary way to have spawnable hard-coded DroppedMechanismComponent prototypes
+            //In the future (when it becomes possible) DroppedMechanismComponent should be auto-generated from the Mechanism prototypes
+            string debugLoadMechanismData = "";
+            base.ExposeData(serializer);
+            serializer.DataField(ref debugLoadMechanismData, "debugLoadMechanismData", "");
+            if (serializer.Reading && debugLoadMechanismData != "")
+            {
+                _prototypeManager.TryIndex(debugLoadMechanismData, out MechanismPrototype data);
+                InitializeDroppedMechanism(new Mechanism(data));
+            }
+        }
+
+
+
+        private void SendBodyPartListToUser(AfterInteractEventArgs eventArgs, BodyManagerComponent bodyManager)
+        {
+            var toSend = new Dictionary<string, object>(); //Create dictionary to send to client (text to be shown : data sent back if selected)
+            foreach (var (key, value) in bodyManager.PartDictionary)
+            { //For each limb in the target, add it to our cache if it is a valid option.
+                if (value.CanInstallMechanism(ContainedMechanism))
+                {
+                    int randomKey = _random.Next(Int32.MinValue, Int32.MaxValue);
```

Comment:
Now that I see why you need random IDs - this is normally done with increasing "next ID" variable, which you just add 1 to every time you want a fresh ID (with wrapping rather than overflow).  Doing that gives more reasonable and deterministic collision behaviour, and isn't vulnerable to the birthday problem (though that isn't a big issue with `int`). It's also a bit simpler to read.

Reply:
Would you just want me to do that with a static integer then? Easy but... seems wrong, somehow.

Reply:
static int is fine.